src/utils/work-time-data-utils.tsx

```diff
@@ -0,0 +1,137 @@
+import { TimeEntry, TimeEntryTotalDto } from "generated/client";
+import strings from "localization/strings";
+import moment from "moment";
+import { WorkTimeCategory, WorkTimeData, WorkTimeDatas, WorkTimeTotalData } from "types";
+
+/**
+ * Utility class for time values
+ */
+export default class WorkTimeDataUtils {
+  /**
+   * Proprocess the date entries for graphs
+   * 
+   * @param dateEntries sorted date entries from api request 
+   * @return processed work time data, work time total data
+   */
+  public static dateEntriesPreprocess = (dateEntries: TimeEntry[]): WorkTimeDatas => {
+    const workTimeData: WorkTimeData[] = [];
+    const workTimeTotalData: WorkTimeTotalData = {
+      name: WorkTimeCategory.TOTAL,
+      total: 0,
+      logged: 0,
+      expected: 0
+    };
+
+    dateEntries.forEach(
+      entry => {
+        workTimeData.push({
+          name: entry.date.toISOString().split("T")[0],
+          expected: entry.expected,
+          project: entry.projectTime,
+          internal: entry.internalTime
+        });
+        workTimeTotalData.total = workTimeTotalData.total + entry.total;
+        workTimeTotalData.logged = workTimeTotalData.total + entry.logged;
+        workTimeTotalData.expected = workTimeTotalData.total + entry.expected;
+      }
+    );
+
+    return { workTimeData, workTimeTotalData };
+  }
+
+  /**
+   * Proprocess the week entries for graphs
+   * 
+   * @param weekEntries sorted week entries from api request 
+   * @return processed work time data, work time total data
+   */
+  public static weekEntriesPreprocess = (weekEntries: TimeEntryTotalDto[]): WorkTimeDatas => {
+    const workTimeData: WorkTimeData[] = [];
+    const workTimeTotalData: WorkTimeTotalData = {
+      name: WorkTimeCategory.TOTAL,
+      total: 0,
+      logged: 0,
+      expected: 0
+    };
+
+    weekEntries.forEach(
+      entry => {
+        workTimeData.push({
+          name: `${entry.id?.year!} ${strings.week} ${entry.id?.week!}`,
+          expected: entry.expected,
+          project: entry.projectTime,
+          internal: entry.internalTime
+        });
+        workTimeTotalData.total = workTimeTotalData.total + entry.total;
+        workTimeTotalData.logged = workTimeTotalData.total + entry.logged;
+        workTimeTotalData.expected = workTimeTotalData.total + entry.expected;
+      }
+    );
+
+    return { workTimeData, workTimeTotalData };
+  }
```

Comment:
This and the last 2 functions seem to do more or less the same thing with just a different name for the entry. So I'd combine these 3 functions to single function, pass extra scope param to it and add another function that gives the correct entry name based on the scope

```suggestion
  /**
   * Preprocess the week, month and year entries for graphs
   * 
   * @param weekEntries sorted week, month or year entries from api request 
   * @param scope filter scope
   * @return processed work time data, work time total data
   */
  public static weeksYearsAndMonthsPreprocess = (weekEntries: TimeEntryTotalDto[], scope: FilterScopes): WorkTimeDatas => {
    const workTimeData: WorkTimeData[] = [];
    const workTimeTotalData: WorkTimeTotalData = {
      name: WorkTimeCategory.TOTAL,
      total: 0,
      logged: 0,
      expected: 0
    };

    weekEntries.forEach(
      entry => {
        workTimeData.push({
          name: WorkTimeDataUtils.getTimeDataName(entry, scope),
          expected: entry.expected,
          project: entry.projectTime,
          internal: entry.internalTime
        });
        workTimeTotalData.total = workTimeTotalData.total + entry.total;
        workTimeTotalData.logged = workTimeTotalData.total + entry.logged;
        workTimeTotalData.expected = workTimeTotalData.total + entry.expected;
      }
    );

    return { workTimeData, workTimeTotalData };
  }

  /**
   * Get time data name
   *
   * @param entry entry
   * @param scope filter scope
   * @returns time data name
   */
  private static getTimeDataName = (entry: TimeEntryTotalDto, scope: FilterScopes): string => {
    if (!entry.id) {
      return ""
    }

    return {
      [FilterScopes.DATE]: "",
      [FilterScopes.WEEK]: `${entry.id?.year!} ${strings.week} ${entry.id?.week!}`,
      [FilterScopes.MONTH]: `${entry.id?.year!}-${entry.id?.month!}`,
      [FilterScopes.YEAR]: `${entry.id?.year!}`
    }[scope];
  }
```